dev/generate-bytebuffer-multi-int.sh

```diff
@@ -0,0 +1,167 @@
+    echo "        guard self.readableBytes >= bytesRequired else {"
+    echo "            return nil"
+    echo "        }"
+    echo
+    for n in $(seq 1 "$how_many"); do
+        echo "        var v$n: T$n = 0"
+    done
+    echo "        var offset = 0"
+    echo "        self.readWithUnsafeReadableBytes { ptr -> Int in"
+    echo "            assert(ptr.count >= bytesRequired)"
+    echo "            let basePtr = ptr.baseAddress! // safe, ptr is non-empty"
+    for n in $(seq 1 "$how_many"); do
+        echo "            withUnsafeMutableBytes(of: &v$n) { destPtr in"
+        echo "                destPtr.baseAddress!.copyMemory(from: basePtr + offset, byteCount: MemoryLayout<T$n>.size)"
+        echo "            }"
+        echo "            offset = offset &+ MemoryLayout<T$n>.size"
+    done
+    echo "            return offset"
+    echo "        }"
+    echo "        switch endianness {"
+    for endianness in big little; do
+        echo "        case .$endianness:"
+        echo -n "            return (T1(${endianness}Endian: v1)"
+        for n in $(seq 2 "$how_many"); do
+            echo -n ", T$n(${endianness}Endian: v$n)"
+        done
+        echo ")"
+    done
+    echo "        }"
+    echo "    }"
+    echo
+
+    # WRITE
+    echo "    @inlinable"
+    echo "    @_alwaysEmitIntoClient"
+    echo "    @discardableResult"
+    echo -n "    public mutating func writeMultipleIntegers<T1: FixedWidthInteger"
+    for n in $(seq 2 "$how_many"); do
+        echo -n ", T$n: FixedWidthInteger"
+    done
+    echo -n ">(_ value1: T1"
+    for n in $(seq 2 "$how_many"); do
+        echo -n ", _ value$n: T$n"
+    done
+    echo -n ", endianness: Endianness = .big, as: (T1"
+    for n in $(seq 2 "$how_many"); do
+        echo -n ", T$n"
+    done
+    echo -n ").Type = (T1"
+    for n in $(seq 2 "$how_many"); do
+        echo -n ", T$n"
+    done
+    echo ").self) -> Int {"
+    for n in $(seq 1 "$how_many"); do
+        echo "        var v$n: T$n"
+    done
+    echo "        switch endianness {"
+    for endianness in .big .little; do
+        echo "        case $endianness:"
+        for n in $(seq 1 "$how_many"); do
+            echo "            v$n = value$n${endianness}Endian"
+        done
+    done
+    echo "        }"
+    echo
+    echo "        var spaceNeeded: Int = MemoryLayout<T1>.size"
+    for n in $(seq 2 "$how_many"); do
+        echo "        spaceNeeded &+= MemoryLayout<T$n>.size"
+    done
+    echo
+    echo "        return self.writeWithUnsafeMutableBytes(minimumWritableBytes: spaceNeeded) { ptr -> Int in"
+    echo "            var offset = 0"
+    echo "            let basePtr = ptr.baseAddress! // safe: pointer is non zero length"
+    for n in $(seq 1 "$how_many"); do
+        echo "            (basePtr + offset).copyMemory(from: &v$n, byteCount: MemoryLayout<T$n>.size)"
+        echo "            offset = offset &+ MemoryLayout<T$n>.size"
+    done
+    echo "            assert(offset <= ptr.count)"
```

Comment:
@Lukasa thanks, I now added `assert`s after each `offset` increment. Is this what you were looking for?

Reply:
Oh, I'm sorry, I should have been clearer. We know how big this has to be, because we have a `minimumWritableBytes` argument above. We should assert that the pointer has that much space before we start writing to it.

Reply:
okay, sounds good! will do

Reply:
@Lukasa done: I now asserted that there's enough space. And additionally that `offset` at the end matches `bytesRequired`/`spaceNeeded`. Those should line up exactly.